src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp

```diff
@@ -1993,27 +2002,33 @@ void Hydro_HancockPredict( real fc[][NCOMP_LR], const real dt, const real dh,
       if ( fc[f][4] <= (real)0.0 || fc[f][4] >= HUGE_NUMBER || fc[f][4] != fc[f][4] ) reset_cell = true;
       if ( Pres     <= (real)0.0 || Pres     >= HUGE_NUMBER || Pres     != Pres     ) reset_cell = true;
```

Comment:
If we passed the primitive variables into `Hydro_HancockPredict`, we should not calculate the `Pres` in the function once more at line 1999.